File: app/src/main/java/com/projecthub/repository/ProjectRepository.java
package com.projecthub.repository;

import com.projecthub.model.Project;
import java.util.List;
import java.util.Optional;
import java.util.UUID;

/**
 * Repository interface for {@link Project} entities.
 */

public interface ProjectRepository {
    Project save(Project project);
    Optional<Project> findById(UUID id);
    List<Project> findAll();
    void delete(Project project);
    List<Project> findAllByTeamId(UUID teamId);
    boolean existsById(UUID projectId);
    Project getReferenceById(UUID projectId);
    List<Project> findByTeamId(UUID teamId);
    void deleteById(UUID id);
}

File: src/main/kotlin/com/projecthub/domain/project/ProjectRepository.kt
package com.projecthub.domain.project

/**
 * Repository interface for projects (outbound port)
 * This interface defines the persistence operations available for Project entities
 */
interface ProjectRepository {
    fun findById(id: String): Project?
    fun findAll(): List<Project>
    fun save(project: Project): Project
    fun deleteById(id: String)
    fun findByOwnerId(ownerId: String): List<Project>
    fun findByTeamId(teamId: String): List<Project>
}
